pycbc/inference/distributions.py

```diff
@@ -1219,6 +1253,195 @@ def from_config(cls, cp, section, variable_args):
         return _bounded_from_config(cls, cp, section, variable_args,
             bounds_required=False)
 
+class FromFile(_BoundedDist):
+    """A distribution that reads the values of the parameter(s) from a file,
+    computes the kde to construct the pdf, and draws random variables from it.
+
+    Parameters
+    ----------
+    file_name : str
+        The path to a file containing the values of the parameters that want
+        to be used to construct the distribution.
+    \**params :
+        The keyword arguments should provide the names of the parameters to be
+        read from the file and their bounds.
+
+    Attributes
+    ----------
+    name : 'fromfile'
+        The name of the distribution.
+    file_name : str
+        The path to the file containing values for the parameter(s).
+    params : list
+        Parameters read from file.
+    norm : float
+        The normalization of the multi-dimensional pdf.
+    lognorm : float
+        The log of the normalization.
+    kde :
+        The kde obtained from the values in the file.
+    """
+    name = 'fromfile'
+    def __init__(self, file_name=None, **params):
+        if file_name is None:
+            raise ValueError('A file must be specified for this distribution.')
+        self._filename = file_name
+        # Get the parameter names to pass to get_kde_from_file
+        if len(params) == 0:
+            ps = None
+        else:
+            ps = params.keys()
+        pnames, self._kde = get_kde_from_file(file_name, params=ps)
+        # If no parameters where given, populate with pnames
+        for param in pnames:
+            if param not in params:
+                params[param] = None
+        super(FromFile, self).__init__(**params)
+        # Make sure to store parameter names in same order as given by kde function
+        self._params = pnames
+        # Compute the norm and save
+        lower_bounds = [self.bounds[p][0] for p in pnames]
+        higher_bounds = [self.bounds[p][1] for p in pnames]
+        # Avoid inf because of inconsistencies in integrate_box
+        for ii, bnd in enumerate(lower_bounds):
+            if abs(bnd) == numpy.inf:
+                lower_bounds[ii] = numpy.sign(bnd) * 2 ** 31
```

Comment:
This is because if you do not specify boundaries, then the default is (-inf, inf), but scipy.stats.gaussian_kde.integrate_box(-inf, inf) returns NAN and therefore this wouldn't broke. So it is hard-coded to a big non-infinity number only for the case that the user doesn't specify boundaries, but there is the option to specify boundaries.